RoosterPlanner.UI/src/app/components/add-shifts/add-shifts.component.ts

```diff
@@ -0,0 +1,216 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {Location} from '@angular/common';
+import {Project} from "../../models/project";
+import {Task} from "../../models/task";
+import {FormBuilder, FormControl, Validators} from "@angular/forms";
+import {MultipleDatesComponent} from "ngx-multiple-dates";
+import {TaskService} from "../../services/task.service";
+import {ProjectService} from "../../services/project.service";
+import {ActivatedRoute, ParamMap} from "@angular/router";
+import {ToastrService} from "ngx-toastr";
+import {ShiftService} from "../../services/shift.service";
+import {Moment} from "moment";
+import {DateConverter} from "../../helpers/date-converter";
+import {Shift} from "../../models/shift";
+import {EntityHelper} from "../../helpers/entity-helper";
+import * as moment from "moment";
+
+@Component({
+  selector: 'app-add-shifts',
+  templateUrl: './add-shifts.component.html',
+  styleUrls: ['./add-shifts.component.scss']
+})
+export class AddShiftsComponent implements OnInit {
+  guid: string;
+  loaded: boolean = false;
+  title: string = "Shift toevoegen";
+
+  project: Project;
+  tasks: Task[];
+  requiredParticipants: number;
+
+  checkoutForm;
+  taskControl: FormControl;
+  startTimeControl: FormControl;
+  endTimeControl: FormControl;
+  participantsRequiredControl: FormControl;
+  daySelectionControl: FormControl;
+  selectionOptions: string[] = ['Elke dag', 'Elke maandag', 'Elke dinsdag', 'Elke woensdag', 'Elke donderdag', 'Elke vrijdag', 'Elke zaterdag', 'Elke zondag', 'Verwijder selectie'];
```

Comment:
Zou een centraal component (service) maken die alle teksten heeft en die injecteren in de ctor.